sdk/storage/inc/datalake/protocol/datalake_rest_client.hpp

```diff
@@ -639,77 +644,69 @@ namespace Azure { namespace Storage { namespace DataLake {
   {
     std::unique_ptr<Azure::Core::Http::BodyStream> BodyStream;
     std::string AcceptRanges;
-    std::string CacheControl;
-    std::string ContentDisposition;
-    std::string ContentEncoding;
-    std::string ContentLanguage;
+    DataLakeHttpHeaders HttpHeaders;
     int64_t ContentLength = int64_t();
-    std::string ContentRange;
-    std::string ContentType;
-    std::string ContentMD5;
+    Azure::Core::Nullable<std::string> ContentRange;
+    Azure::Core::Nullable<std::string> ContentMD5;
     std::string Date;
     std::string ETag;
     std::string LastModified;
     std::string RequestId;
     std::string Version;
     std::string ResourceType;
-    std::string Properties;
-    std::string LeaseDuration;
-    std::string LeaseState;
-    std::string LeaseStatus;
-    std::string XMsContentMd5;
+    Azure::Core::Nullable<std::string> Properties;
+    Azure::Core::Nullable<std::string> LeaseDuration;
+    Azure::Core::Nullable<std::string> LeaseState;
+    Azure::Core::Nullable<std::string> LeaseStatus;
+    Azure::Core::Nullable<std::string> XMsContentMd5;
   };
 
   struct PathGetPropertiesResponse
   {
-    std::string AcceptRanges;
-    std::string CacheControl;
-    std::string ContentDisposition;
-    std::string ContentEncoding;
-    std::string ContentLanguage;
+    Azure::Core::Nullable<std::string> AcceptRanges;
+    DataLakeHttpHeaders HttpHeaders;
     int64_t ContentLength = int64_t();
-    std::string ContentRange;
-    std::string ContentType;
-    std::string ContentMD5;
+    Azure::Core::Nullable<std::string> ContentRange;
+    Azure::Core::Nullable<std::string> ContentMD5;
     std::string Date;
     std::string ETag;
     std::string LastModified;
     std::string RequestId;
     std::string Version;
-    std::string ResourceType;
-    std::string Properties;
-    std::string Owner;
-    std::string Group;
-    std::string Permissions;
-    std::string ACL;
-    std::string LeaseDuration;
-    std::string LeaseState;
-    std::string LeaseStatus;
+    Azure::Core::Nullable<std::string> ResourceType;
+    Azure::Core::Nullable<std::string> Properties;
+    Azure::Core::Nullable<std::string> Owner;
+    Azure::Core::Nullable<std::string> Group;
+    Azure::Core::Nullable<std::string> Permissions;
+    Azure::Core::Nullable<std::string> ACL;
+    Azure::Core::Nullable<std::string> LeaseDuration;
+    Azure::Core::Nullable<std::string> LeaseState;
+    Azure::Core::Nullable<std::string> LeaseStatus;
```

Comment:
Both status and state should be enum

Reply:
This is another bug in current DataLake swagger that's not consistent with blob's swagger, sigh.
Modified the swagger to make this work.